lib/contracts/builtin_contracts.rb

```diff
@@ -385,9 +385,10 @@ def initialize(options)
     end
 
     def valid?(hash)
-      options.all? do |key, contract|
-        Optional._valid?(hash, key, contract)
-      end
+      (hash.keys - options.keys == []) &&
```

Comment:
Can you use guard statement here instead - current code makes it hard to read. Example:

```ruby
def valid?(hash)
  return false unless hash.keys - options.keys == []      # <--- this is a guard statement
  # .. do the main job here ..
end
```